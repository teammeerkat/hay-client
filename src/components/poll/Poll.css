.poll{
  display: inline-block;
  background: red;
  padding: 2%;
}

.poll li{
  border-bottom: 1px solid white;
  margin-bottom: 1em;

}

.poll li:hover{
  background: maroon;
}