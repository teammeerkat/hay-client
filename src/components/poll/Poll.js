import React from 'react';
import PropTypes from 'prop-types';
import styles from './Poll.css';

export default function Poll({ question, options }) {
  const optionsList = options.map((option, i) => {
    return <li key={i}>{option}</li>;
  });
  return (
    <section className={styles.poll}>
      <h2>{question}</h2> 
      <ul>{optionsList}</ul>
      
    </section>
  );
}

Poll.propTypes = {
  question: PropTypes.string.isRequired,
  options: PropTypes.array.isRequired
};
