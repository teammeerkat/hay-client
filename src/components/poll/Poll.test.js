import React from 'react';
import renderer from 'react-test-renderer';
import Poll from './Poll';
describe('Polls Test', () => {
  it('Renders list of polls', () => {

    const tree = renderer.create(
      <Poll question={'some Question'} options={['option1', 'option2']} /> 
    );
    expect(tree).toMatchSnapshot();
  });
});
