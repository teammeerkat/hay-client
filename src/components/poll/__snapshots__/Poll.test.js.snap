// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Polls Test Renders list of polls 1`] = `
<section
  className="poll"
>
  <h2>
    some Question
  </h2>
  <ul>
    <li>
      option1
    </li>
    <li>
      option2
    </li>
  </ul>
</section>
`;
